### DISCUSSION TITLE
An error occurred while fetching your account

### DISCUSSION THREAD

**Discussion by richleach** [payloadcms/payload#5023] [Q&A]
### Link to reproduction

https://thebeatentrail-supabase.vercel.app/

### Describe the Bug

I created a site deployment on Vercel for my Payload CMS application - everything has tested OK/passed on my dev server. 
I received confirmation that my site built successfully - no errors. When I click the link https://thebeatentrail-supabase.vercel.app/ the site loads (my headers/footers/nav are all there) but any page that has content from the /admin is not loading, I get a 404 page displayed (by design). The console displays....

Uncaught (in promise) Error: An error occurred while fetching your account.

When I drill down into the error message I can see the following:

 n.useEffect)(()=>{
                let e = async()=>{
                    try {
                        let e = await fetch("".concat("https://thebeatentrail-supabase.vercel.app", "/api/users/me"), {
                            method: "GET",
                            credentials: "include",
                            headers: {
                                "Content-Type": "application/json"
                            }
                        });
                        if (e.ok) {
                            let {user: t} = await e.json();
                            l(t || null),
                            s(t ? "loggedIn" : void 0)
                        } else
                            throw Error("An error occurred while fetching your account.")
                    } catch (e) {
                        throw l(null),
                        Error("An error occurred while fetching your account.")
                    }
                };

### To Reproduce

I visit the link https://thebeatentrail-supabase.vercel.app/

.... the homepage, as an example, and even the page /terms (are pages I built using the /admin console of the CMS) will not load and the corresponding 404 messages are displayed.

The page /videos is a hard-coded file served from the file system (not my Supabase database). This page loads successfully.

### Payload Version

"payload": "^2.0.0",

### Adapters and Plugins

db: postgresAdapter({
    pool: {
      connectionString: process.env.DATABASE_URI,
    },
    push: false,
  }),

**Comment by richleach** [comment]
To reproduce this error: 
Make sure you're on version "payload": "^2.0.0". (view in package.json)
I run the payload web app locally on my server - command line "npm run dev".
After building the Payload app locally, and creating at least one page via the /admin, I verified that the page displayed, as expected in my local browser (localhost:3000).
I checked in my files into my github repo, on the main branch.
I logged into my vercel account at vercel.com.
I created a deployment following the instructions to point my deployment to the appropriate git branch.
I manually created (and double checked) the environment variables in that section of the Vercel site of my deployment.
I created the deployment, with no errors being reported from the Vercel build report nor the log files.
I open a new browser and enter the url given to me as assigned by the Vercel DNS for my project.
The first request loads the shell/files from the file server so my application is visible (headers/footers/nav bar are all present). A 404 error is displayed.
In the console is the reported error:

Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'attribution-reporting'.
 
        
        
GET https://thebeatentrail-supabase.vercel.app/ 404 (Not Found)
C @ 864-6d7511f2f6e1717f.js:1
ro @ fd9d1056-35ddd8600555f926.js:9
lp @ fd9d1056-35ddd8600555f926.js:9
om @ fd9d1056-35ddd8600555f926.js:9
iD @ fd9d1056-35ddd8600555f926.js:9
(anonymous) @ fd9d1056-35ddd8600555f926.js:9
(anonymous) @ fd9d1056-35ddd8600555f926.js:9
ib @ fd9d1056-35ddd8600555f926.js:9
M @ 864-6d7511f2f6e1717f.js:25
thebeatentrail-supabase.vercel.app/:1 Refused to load the font 'data:font/woff2;base64,d09GMgABAAAAAQ5sABIAAAADuOQAAQ4EAAEAgwAAAAAAAAAAAAAAAAAAAAAAAAAAG4SpIhy4XhSHMgZgAJJmCF4JgnMREAqGskiFyggSgrVoATYCJAOdOAuOXgAEIAWPGAfGYAyDGFsKfLME/nOMvX+HnQoMqyRqK6tt2TW8K2BDX9Umu4PZkCXwtMjpcHYNTx6UUmu30UdwcxxnVMvcNgAim7rdXiPZ////////////C5dF2ObuXM7ZvQeSECARwueHlaLVtkJEIaLRAI3ulswSE13cSPdM15RCyYQuuS7A0mBqpQO9pYZWQQMqa9lQb0dpJzHxHpLyCHIi8uBdwF58nNH+YOA40OpJvEjJtaokKaUkx/N5UeSinp5TWoW1vSpFjimBgeFGeCF66U0n6ZUN/qSqC4XC11GKUNmzSy2d5Jxyqw5JzGUlJUnyi0lKkE8BlxadTZsyRm8c1/Cti2hEjzIHnuOZ0fU6OjKc...FVw9csGnQ1BnbMbfzgRj3WjPytwZYWpPh3OtmsLZaNcxxqaGVqMVqFffFpU8bbs/0N1vG3POWfS7LUeB81TfHn+VbD7x/UnT3lXyFltHfdX6+X6Rzmg9d4dAISK7GlTj2zgME/JUwAETAECwABAUgUmAysrKI5shRwAchnFSMheQ8UoKlvnVbkW9mMeq2+bbO9EQ2q09Z9wzsADarYt4ftoamBDi8gmfbZKnl+lkZWyRx1VYVSIpVFXs9VXy9ytCyTNYt09YIAGKjqpDT22SgOdP5OsdAK/vOCXFnrpEwA/PPTJiopB7HxZV+MlaGCT/04K/LY3kjPYV/YR146Ggg+9sX6na/c2So+Epz3PXve3PKRylPIhNZASBUYVDIGhikiWc1jj/claeVCJvljX7JfvO+xhfTDIjSXBs4HV16Ol/BMfcuFQ4NRmDOWEh6diOI6NZmGl87N9J+r2JfpW2+5NhV/fGHLO57Yavbq7zaCewa/RDxp/aSH+luOPwAA' because it violates the following Content Security Policy directive: "font-src 'self'".

408-09d1499de47de96d.js:1 
        
        
       GET https://thebeatentrail-supabase.vercel.app/api/users/me 404 (Not Found)
e @ 408-09d1499de47de96d.js:1
(anonymous) @ 408-09d1499de47de96d.js:1
ap @ fd9d1056-35ddd8600555f926.js:9
aM @ fd9d1056-35ddd8600555f926.js:9
(anonymous) @ fd9d1056-35ddd8600555f926.js:9
M @ 864-6d7511f2f6e1717f.js:25
408-09d1499de47de96d.js:1 Uncaught (in promise) Error: An error occurred while fetching your account.
    at e (408-09d1499de47de96d.js:1:2824)
e @ 408-09d1499de47de96d.js:1
await in e (async)
(anonymous) @ 408-09d1499de47de96d.js:1
ap @ fd9d1056-35ddd8600555f926.js:9
(anonymous) @ fd9d1056-35ddd8600555f926.js:9
M @ 864-6d7511f2f6e1717f.js:25

Clicking onto the error reveals this code as the offending error generator:
n.useEffect)(()=>{
                let e = async()=>{
                    try {
                        let e = await fetch("".concat("https://thebeatentrail-supabase.vercel.app", "/api/users/me"), {
                            method: "GET",
                            credentials: "include",
                            headers: {
                                "Content-Type": "application/json"
                            }
                        });
                        if (e.ok) {
                            let {user: t} = await e.json();
                            l(t || null),
                            s(t ? "loggedIn" : void 0)
                        } else
                            throw Error("An error occurred while fetching your account.")
                    } catch (e) {
                        throw l(null),
                        Error("An error occurred while fetching your account.")
                    }
                }
                ;


...which appears to be running in the file 
/src/app/_providers/Auth

Specifically the line:
const res = await fetch(`${process.env.NEXT_PUBLIC_SERVER_URL}/api/users/me`, { ...

...that reports the 404 when attempting to call the parsed url:
thebeatentrail-supabase.vercel.app/api/users/me

I hope this helps and get us closer to a resolution.

Rich

**Comment by richleach** [comment]
@AlessioGr I hope these details were helpful, were you able to reproduce this issue? I can be available via chat to walk you through any questions about this. 
Thanks again, Rich

**Comment by richleach** [comment]
In an effort to expedite this issue I was searching for some possible insight, 
![image](https://github.com/payloadcms/payload/assets/1168807/c1412ba5-1806-41b7-be68-162d4716fed2)

appears like a possible config issue/error might be afoot... The screenshot above indicates more than one CORS issue so I thought I'd add one or more of these errors to my whitelist (payload.config.ts)..... Trouble is, the fonts - as I understand Next.js builds - don't they download the fonts at build time so that you don't have to request them at page load time? And there's even a stripe.com call getting a CORS flag, and that's already in the cors: and csrf: sections.... so perhaps my updated question now is how would I deal with solving these CORS errors in Payload? I ask all that to get to the question of should I add "/api/users/me" to the cors: and csrf: sections so that Payload/Next.js let's it run?

**Comment by richleach** [comment]
@AlessioGr  I updated my payload.config.ts file to include a few sites in the csrf array to reflect the possible domains that Vercel dynamically assigned during their build process. Not only did that not improve this problem it actually spawned a few more warnings:
Error with Permissions-Policy header: Origin trial controlled feature not enabled: 'attribution-reporting'.
admin:1 
        
        
       GET https://thebeatentrail-supabase-jhico4t77-michael-herrmanns-projects.vercel.app/admin 404 (Not Found)
admin:1 Refused to load the font 'data:font/woff2;base64,d09GMgABAAAAAQ5sABIAAAADuOQAAQ4EAAEAgwAAAAAAAAAAAAAAAAAAAAAAAAAAG4SpIhy4XhSHMgZgAJJmCF4JgnMREAqGskiFyggSgrVoATYCJAOdOAuOXgAEIAWPGAfGYAyDGFsKfLME/nOMvX+HnQoMqyRqK6tt2TW8K2BDX9Umu4PZkCXwtMjpcHYNTx6UUmu30UdwcxxnVMvcNgAim7rdXiPZ////////////C5dF2ObuXM7ZvQeSECARwueHlaLVtkJEIaLRAI3ulswSE13cSPdM15RCyYQuuS7A0mBqpQO9pYZWQQMqa9lQb0dpJzHxHpLyCHIi8uBdwF58nNH+YOA40OpJvEjJtaokKaUkx/N5UeSinp5TWoW1vSpFjimBgeFGeCF66U0n6ZUN/qSqC4XC11GKUNmzSy2d5Jxyqw5JzGUlJUnyi0lKkE8BlxadTZsyRm8c1/Cti2hEjzIHnuOZ0fU6OjKc...FVw9csGnQ1BnbMbfzgRj3WjPytwZYWpPh3OtmsLZaNcxxqaGVqMVqFffFpU8bbs/0N1vG3POWfS7LUeB81TfHn+VbD7x/UnT3lXyFltHfdX6+X6Rzmg9d4dAISK7GlTj2zgME/JUwAETAECwABAUgUmAysrKI5shRwAchnFSMheQ8UoKlvnVbkW9mMeq2+bbO9EQ2q09Z9wzsADarYt4ftoamBDi8gmfbZKnl+lkZWyRx1VYVSIpVFXs9VXy9ytCyTNYt09YIAGKjqpDT22SgOdP5OsdAK/vOCXFnrpEwA/PPTJiopB7HxZV+MlaGCT/04K/LY3kjPYV/YR146Ggg+9sX6na/c2So+Epz3PXve3PKRylPIhNZASBUYVDIGhikiWc1jj/claeVCJvljX7JfvO+xhfTDIjSXBs4HV16Ol/BMfcuFQ4NRmDOWEh6diOI6NZmGl87N9J+r2JfpW2+5NhV/fGHLO57Yavbq7zaCewa/RDxp/aSH+luOPwAA' because it violates the following Content Security Policy directive: "font-src 'self'".

408-09d1499de47de96d.js:1 Refused to connect to 'https://thebeatentrail-supabase.vercel.app/api/users/me' because it violates the following Content Security Policy directive: "connect-src 'self' https://checkout.stripe.com https://api.stripe.com https://maps.googleapis.com".

e @ 408-09d1499de47de96d.js:1
(anonymous) @ 408-09d1499de47de96d.js:1
ap @ fd9d1056-35ddd8600555f926.js:9
(anonymous) @ fd9d1056-35ddd8600555f926.js:9
M @ 864-6d7511f2f6e1717f.js:25
408-09d1499de47de96d.js:1 Refused to connect to 'https://thebeatentrail-supabase.vercel.app/api/users/me' because it violates the document's Content Security Policy.
e @ 408-09d1499de47de96d.js:1
(anonymous) @ 408-09d1499de47de96d.js:1

iB @ fd9d1056-35ddd8600555f926.js:9
(anonymous) @ fd9d1056-35ddd8600555f926.js:9
M @ 864-6d7511f2f6e1717f.js:25
408-09d1499de47de96d.js:1 Uncaught (in promise) Error: An error occurred while fetching your account.
    at e (408-09d1499de47de96d.js:1:2824)
e @ 408-09d1499de47de96d.js:1
await in e (async)
(anonymous) @ 408-09d1499de47de96d.js:1
ap @ fd9d1056-35ddd8600555f926.js:9

(anonymous) @ fd9d1056-35ddd8600555f926.js:9
M @ 864-6d7511f2f6e1717f.js:25
admin:1 The resource https://thebeatentrail-supabase-jhico4t77-michael-herrmanns-projects.vercel.app/_next/static/css/e8484a29af24051f.css was preloaded using link preload but not used within a few seconds from the window's load event. Please make sure it has an appropriate `as` value and it is preloaded intentionally.

If the consistently reported issue is always '/api/users/me' - which appears to be an authentication file, is there a possible missing step during installation (or, in my case during deployment on Vercel)?

**Comment by jmikrut** [comment]
Hey @richleach - this seems like a problem with your setup, rather than Payload itself, so I am going to convert this into a Discussion and we can help assist you there.

But I think in order for us to be helpful here, we really will need a minimal, reproducible example. 

And I see that you have a lot of different issues / questions here so I would say in order for us to help you as easily as possible, we should try and focus on one thing at a time.

One thing from me is that by looking at this for a quick second, I see that you are trying to load `/api/users/me` from your Vercel-deployed website.

```
   GET https://thebeatentrail-supabase.vercel.app/api/users/me 404 (Not Found)
```

I don't see that you are running Payload on your Vercel website, so that endpoint is 404'ing.

**Comment by richleach** [comment]
James
Thanks for jumping in.
I actually started this application from the Payload ecommerce github repo (I didn't realize at the time I was going down the Cloud path, so I followed the links to clone that repo after signing up with Payload). Once I got that codebase onto my local server I simply developed it as any Next.js application. I created a Supabase / Postgresql account and housed my data in that DB. Most of my activity focused on styling and layout, and later on I connected a Youtube account to Payload so that I could integrate movie data onto one of my pages. All of this ran fine and generated no errors on my localhost:3000, but when I built my site on Vercel I got my completed and styled site to appear but none of the pages that I generated from the /admin section appear; they 404 - but it's Payload's 404 files that are displayed. So at the very least there is some Payload functionality working, just none of the page retrieval from my Posgresql DB. 
I'm happy to correspond with any specifics as I regularly updated my git repo with meaningful comments along the way.
Thanks again for your help.

**Comment by richleach** [comment]
... and unfortunately I got a little over-zealous and opened up more than 1 thread trying to remedy this issue, which is why you'll find other threads with me asking about NEXT_PUBLIC_SERVER_URL and other necessary environment variable settings as I originally suspected a path issue when I saw this error:
GET https://thebeatentrail-supabase.vercel.app/api/users/me 404 (Not Found)

However about that same time others started chiming in to inform me that Payload was never going to work on Vercel's serverless environment and that I needed to deploy on a different environment with a Node server separate from the Payload content....

**Reply by OliveiraDCI** [reply]
really...

**Reply by richleach** [reply]
?
